databricks_cli/instance_pools/cli.py

```diff
@@ -0,0 +1,164 @@
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from json import loads as json_loads
+
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType, JsonClickType, InstancePoolIdClickType
+from databricks_cli.instance_pools.api import InstancePoolsApi
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format, json_cli_base, \
+    truncate_string
+from databricks_cli.configure.config import provide_api_client, profile_option, debug_option
+from databricks_cli.version import print_version_callback, version
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None, type=click.Path(),
+              help='File containing JSON request to POST to /api/2.0/instance_pools/create.')
+@click.option('--json', default=None, type=JsonClickType(),
+              help=JsonClickType.help('/api/2.0/instance_pools/create'))
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_cli(api_client, json_file, json):
+    """
+    Creates a Databricks instance pool.
+
+    The specification for the request json can be found at
+    https://docs.databricks.com/api/latest/instance-pools.html#create
+    """
+    json_cli_base(json_file, json, lambda json: InstancePoolsApi(api_client).
+                  create_instance_pool(json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None, type=click.Path(),
+              help='File containing JSON request to POST to /api/2.0/instance_pools/edit.')
+@click.option('--json', default=None, type=JsonClickType(),
+              help=JsonClickType.help('/api/2.0/instance_pools/edit'))
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def edit_cli(api_client, json_file, json):
+    """
+    Edits a Databricks instance pool.
+
+    The specification for the request json can be found at
+    https://docs.databricks.com/api/latest/instance-pools.html#edit
+    """
+    if not bool(json_file) ^ bool(json):
+        raise RuntimeError('Either --json-file or --json should be provided')
+    if json_file:
+        with open(json_file, 'r') as f:
+            json = f.read()
+    deser_json = json_loads(json)
+    InstancePoolsApi(api_client).edit_instance_pool(deser_json)
```

Comment:
Why can't this be:

```
json_cli_base(json_file, json, 
              lambda json: InstancePoolsApi(api_client).edit_instance_pool(json))
```
Seems like we are duplicating code in `json_cli_base`. I see that `clusters/cli.py` does this for edit as well. Is there are reason?

Reply:
I mimic the cluster cli:
```
@debug_option
@profile_option
@eat_exceptions
@provide_api_client
def edit_cli(api_client, json_file, json):
    """
    Edits a Databricks cluster.

    The specification for the request json can be found at
    https://docs.databricks.com/api/latest/clusters.html#edit
    """
    if not bool(json_file) ^ bool(json):
        raise RuntimeError('Either --json-file or --json should be provided')
    if json_file:
        with open(json_file, 'r') as f:
            json = f.read()
    deser_json = json_loads(json)
    ClusterApi(api_client).edit_cluster(deser_json)
```
@andrewmchen, do you know the reason behind? Thank you!